File: python/examples/read_profile.py
#!/usr/bin/env python3
"""
read_profile.py - Example: read and inspect an ICC profile header.

Usage:
    python read_profile.py [profile.icc]
"""

from pathlib import Path
import sys
import iccdev


def find_example_profile() -> str:
    repo_root = Path(__file__).resolve().parents[2]
    candidates = [
        repo_root / "Testing" / "Display" / "sRGB_D65_MAT.icc",
        repo_root / "Testing" / "Display" / "sRGB_D65_colorimetric.icc",
        repo_root / "Testing" / "Display" / "LCDDisplay.icc",
    ]
    for path in candidates:
        if path.is_file():
            return str(path)
    raise FileNotFoundError(
        "No bundled display profile found. Run Testing/CreateAllProfiles.sh first."
    )


def main():
    path = sys.argv[1] if len(sys.argv) >= 2 else find_example_profile()

    with iccdev.IccProfile(path) as profile:
        header = profile.header

        print(f"Profile: {path}")
        print(f"  Version:      {header.version_string}")
        print(f"  Size:         {header.size} bytes")
        print(f"  Device class: {header.device_class_name}")
        print(f"  Color space:  {header.color_space_name}")
        print(f"  PCS:          {header.pcs_name}")
        print(f"  Platform:     {iccdev.sig_to_str(header.platform)}")
        print(f"  CMM:          {iccdev.sig_to_str(header.cmm_id)}")
        print(f"  Creator:      {iccdev.sig_to_str(header.creator)}")
        print(f"  Intent:       {header.rendering_intent_name}")
        print(f"  Illuminant:   X={header.illuminant['X']:.4f} "
              f"Y={header.illuminant['Y']:.4f} "
              f"Z={header.illuminant['Z']:.4f}")
        print(f"  Date:         {header.date['year']}-{header.date['month']:02d}-"
              f"{header.date['day']:02d} {header.date['hours']:02d}:"
              f"{header.date['minutes']:02d}:{header.date['seconds']:02d}")
        print(f"  Profile ID:   {header.profile_id.hex()}")


if __name__ == "__main__":
    main()

File: matlab/examples/read_profile.m
%% read_profile.m - Read and inspect an ICC profile header
%
% Usage:
%   run this script after building icc_mex and adding matlab/ to path.
%
% Copyright (c) International Color Consortium.
% BSD 3-Clause License. See LICENSE.md for details.

thisDir = fileparts(mfilename('fullpath'));
matlabDir = fileparts(thisDir);
repoRoot = fileparts(matlabDir);
addpath(matlabDir);

candidates = {
  fullfile(repoRoot, 'Testing', 'Display', 'sRGB_D65_MAT.icc')
  fullfile(repoRoot, 'Testing', 'Display', 'sRGB_D65_colorimetric.icc')
  fullfile(repoRoot, 'Testing', 'Display', 'LCDDisplay.icc')
  fullfile(repoRoot, 'Testing', 'sRGB_v4_ICC_preference.icc')
};
profilePath = '';
for i = 1:numel(candidates)
  if exist(candidates{i}, 'file')
    profilePath = candidates{i};
    break;
  end
end
if isempty(profilePath)
  error('iccdev:exampleProfileNotFound', ...
    ['No bundled display profile found. On Windows, run ' ...
     './CreateAllProfiles.bat from Testing in PowerShell; on Unix, run ' ...
     './CreateAllProfiles.sh from Testing.']);
end

fprintf('=== iccdev MATLAB Example: Read Profile ===\n\n');

p = iccdev.IccProfile(profilePath);
hdr = p.header();

fprintf('Profile:      %s\n', profilePath);
fprintf('Version:      %s\n', hdr.versionString);
fprintf('Size:         %d bytes\n', hdr.size);
fprintf('Color Space:  %s (0x%08X)\n', ...
  iccdev.sig_to_str(uint32(hdr.colorSpace)), uint32(hdr.colorSpace));
fprintf('PCS:          %s (0x%08X)\n', ...
  iccdev.sig_to_str(uint32(hdr.pcs)), uint32(hdr.pcs));
fprintf('Device Class: 0x%08X\n', uint32(hdr.deviceClass));
fprintf('Platform:     %s\n', iccdev.sig_to_str(uint32(hdr.platform)));
fprintf('Intent:       %d\n', hdr.renderingIntent);
fprintf('Illuminant:   X=%.4f Y=%.4f Z=%.4f\n', ...
  hdr.illuminantX, hdr.illuminantY, hdr.illuminantZ);
fprintf('Date:         %04d-%02d-%02d %02d:%02d:%02d\n', ...
  hdr.dateYear, hdr.dateMonth, hdr.dateDay, ...
  hdr.dateHours, hdr.dateMinutes, hdr.dateSeconds);
fprintf('Profile ID:   %s\n', sprintf('%02x', hdr.profileId));

p.close();
fprintf('\nDone.\n');
